lib/Parse/Lexer.cpp

```diff
@@ -1558,14 +1668,33 @@ void Lexer::tryLexEditorPlaceholder() {
 }
 
 StringRef Lexer::getEncodedStringSegment(StringRef Bytes,
-                                         SmallVectorImpl<char> &TempString) {
+                                         SmallVectorImpl<char> &TempString,
+                                         unsigned Modifiers,
+                                         const std::string &ToStrip) {
```

Comment:
Replace `Modifiers` and `ToStrip` with explicit boolean that indicate first and last segments, and the number of bytes to skip.